### PR TITLE
ENH: no output in editable verbose mode when there is no work to do

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

**Comment by lesteve** [comment]
I had a quick look and it looks good, thanks for this!

I had some small changes on top of #579 but it looks like your changes are simpler.

File: tests/test_editable.py
Comment (dnicolodi) [reply tests/test_editable.py]: Thanks, fixed.

**Comment by dnicolodi** [comment]
I extended the tests further and improved the log message. I think logging the package name instead of the top level module names is much better. It requires passing one more argument to the loader instance, but I don't see any draw back in doing so.

**Review by ogrisel — APPROVED** [review]
From an end user point of view, the feedback looks good:

```
$ touch sklearn/ensemble/_gradient_boosting.pyx

$ python -c "import sklearn"
meson-python: building scikit-learn: /Users/ogrisel/miniforge3/envs/dev/bin/ninja
[3/3] Linking target sklearn/ensemble/_gradient_boosting.cpython-311-darwin.so

$ python -c "import sklearn"  # no output when no work to do
```

Thanks very much for getting this through.

**Comment by lesteve** [comment]
This one seems ready to be merged, cc @rgommers. Let me know is someone else is likely to review!

Once this PR is merged it would be great if a release of meson-python could be made. There is not a huge rush, but together with https://github.com/mesonbuild/meson-python/pull/569 this would make it easier for scikit-learn to transition to Meson as our main build backend, see https://github.com/scikit-learn/scikit-learn/pull/28506 for more details.

**Comment by dnicolodi** [comment]
The unwritten policy is that "significant" PRs by project maintainers need review by another maintainer. I suspect @rgommers is very busy with the upcoming NumPy 2.0 release. This is not a critical bug fix (from our point of view, at least), thus you'll need to be patient for a bit longer 🙂

**Comment by lesteve** [comment]
Thanks for your answer, and indeed I completely understand that people have other priorities (I should have been more explicit about this in my previous message) !

**Comment by dnicolodi** [comment]
@rgommers After thinking a bit more abut it, I'll be merging this PR before releasing 0.16.0. Unless you object, I'll be doing this before the end of the week.

**Review by rgommers — APPROVED** [review]
Agreed, sorry for the long delay @dnicolodi & all. I have actually been using this PR for a while and everything seems to work like a charm. I had one pending review comment regarding line length, but that is certainly non-blocking and will likely be fine as is. So let's get this in.

Thanks again everyone!

**Comment by lesteve** [comment]
Great to see this one merged, thanks :tada:

**Pull request merged**
